Modules/Core/Common/include/itkNeighborhood.h

```diff
@@ -96,28 +96,20 @@ class ITK_TEMPLATE_EXPORT Neighborhood
   using NeighborIndexType = SizeValueType;
 
   /** Default constructor. */
-  Neighborhood()
-  {
-    m_Radius.Fill(0);
-    m_Size.Fill(0);
-    for (DimensionValueType i = 0; i < VDimension; ++i)
-    {
-      m_StrideTable[i] = 0;
-    }
-  }
+  Neighborhood() = default;
```

Comment:
Could we apply the rule of zero here? Declare no constructor, assignment or move operators?

Reply:
@dzenanz  Thanks for your comment!

> Could we apply the rule of zero here? Declare no constructor, assignment or move operators?

Not entirely _zero_! We must still keep the destructor of `itk::Neighborhood`, as it is virtual. So because we have an explicitly declared destructor, I think it is reasonable to declare all of them (albeit `= default`).